mod/kalvidassign/renderer.php

```diff
@@ -781,21 +781,6 @@ public function display_submissions_table($cm, $groupfilter = 0, $filter = 'all'
                 // No groups, do nothing
                 break;
             case SEPARATEGROUPS:
-                // If separate groups, but displaying all users then we must display only users
-                // who are in the same group as the current user
```

Comment:
@gonenradai - yes, it is intentional.
In our plugin context, we do not differentiate between 'separate groups' and 'visible groups'. According to the (Moodle) documentation:
1. "Visible groups" means that members in other groups can see other groups' assignments.
2. "Separate groups" means that members in other groups can't see other groups' assignments.

@omridevk and I looked around the code in order to see whether we actually enforce the difference between the group modes, and we couldn't find any. This is why we decided to allow the fallback in the switch.
Also, we tested the different scenarios and it looked like it fixed the initial issue + nothing else broke.

@omridevk - if I missed any detail, please share.